lib/core/cache/tweet_cache.dart

```diff
@@ -0,0 +1,62 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:harpy/api/twitter/data/tweet.dart';
+import 'package:harpy/core/app_configuration.dart';
+import 'package:harpy/core/filesystem/cache_dir_service.dart';
+import 'package:logging/logging.dart';
+
+class TweetCache {
+  final Logger log = new Logger('TweetCache');
+
+  CacheDirectoryService _cacheDirService;
+
+  TweetCache() {
+    _cacheDirService = CacheDirectoryService();
+  }
+
+  Future<List<Tweet>> checkCacheForTweets() async {
+    List<Tweet> tweets = [];
+    List<File> files = await _cacheDirService.listFiles(currentBucketName);
+    log.fine("Found ${files.length} cached Tweets!");
+
+    files.forEach((file) {
+      if (_isFileValidForCache(file)) {
+        file.deleteSync();
+      } else {
+        tweets.add(Tweet.fromJson(jsonDecode(file.readAsStringSync())));
+      }
+    });
+
+    return tweets;
+  }
+
+  void cacheTweets(List<Tweet> tweets) {
+    tweets.forEach((tweet) => cacheTweet(tweet));
+  }
+
+  bool _isFileValidForCache(File file) {
+    return file.lastModifiedSync().difference(DateTime.now()).inHours ==
+        AppConfiguration()
+            .applicationConfig
+            .cacheConfiguration
+            .tweetCacheTimeInHours;
+  }
+
+  void cacheTweet(Tweet tweet) {
```

Comment:
sollten wir wirklich jeden tweet in einem file cachen? hat das nicht einen performance einfluss, könnte man es in einer file haben?

Reply:
Hmm good Point. Kann wäre auch kein Problem. Kanns ja mal ausprobieren und wenn es besser ist bau ich es um. Solange würde ich es aber schonmal mergen wollen. Dann kannst du es nutzen

Reply:
ja können es auch immer im nachhinein performance improven, hauptsache es funktioner erstmal